### PR TITLE
REST API: Migrate orders endpoint

### LINKED ISSUE #8550: REST API: Migrate Orders endpoints
As part of the REST API project, we want to migrate Orders to use REST API:

- [ ] Update mappers: Order list and Order item to parse contents without the `data` envelope.
- [ ] Update all Jetpack requests for `OrdersRemote` to enable REST request.

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes #8550 

## Description

This PR updates all mappers related to orders endpoints: `ReportOrderTotalsMapper`, `OrderListMapper`, `OrderMapper`, `OrderNoteMapper` to parse contents without the data envelope.

All endpoints with paths `/orders`, `/notes` and `/reports/orders/totals` are enabled for REST API requests.

## Testing instructions

- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- Log out of the app or skip onboarding if needed.
- On the prologue screen, select Enter your site address and input the address of a self-hosted store.
- Enter correct site credentials to log in.
- When the login succeeds, navigate to the Orders tab. Try creating, updating, loading orders. Everything should work properly.

Please feel free to test again by logging in with a WPCom account. All order-related features should work too.

## Screenshots

N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/NetworkingTests/Mapper/OrderListMapperTests.swift [hunk 1]
```diff
@@ -156,6 +185,12 @@ private extension OrderListMapperTests {
         return mapOrders(from: "orders-load-all")
     }
 
+    /// Returns the OrderListMapper output upon receiving `orders-load-all-without-data`
```
Comment (jaclync) [inline Networking/NetworkingTests/Mapper/OrderListMapperTests.swift]: super nit:
```suggestion
    /// Returns the [Order] output upon receiving `orders-load-all-without-data`
```

File: Networking/NetworkingTests/Mapper/OrderMapperTests.swift [hunk 2]
```diff
@@ -370,6 +402,12 @@ private extension OrderMapperTests {
         return mapOrder(from: "order")
     }
 
+    /// Returns the OrderMapper output upon receiving `order-without-data`
```
Comment (jaclync) [inline Networking/NetworkingTests/Mapper/OrderMapperTests.swift]: super nit:
```suggestion
    /// Returns the Order output upon receiving `order-without-data`
```

File: Networking/NetworkingTests/Mapper/OrderNotesMapperTests.swift [hunk 3]
```diff
@@ -62,12 +87,18 @@ private extension OrderNotesMapperTests {
...
     }
 
-    /// Returns the OrderNotesMapper output upon receiving `orders-load-all`
+    /// Returns the OrderNotesMapper output upon receiving `order-notes`
     ///
     func mapLoadAllOrderNotesResponse() -> [OrderNote] {
         return mapNotes(from: "order-notes")
     }
 
+    /// Returns the OrderNotesMapper output upon receiving `order-notes-without-data`
```
Comment (jaclync) [inline Networking/NetworkingTests/Mapper/OrderNotesMapperTests.swift]: super nit:
```suggestion
    /// Returns the [OrderNote] output upon receiving `order-notes-without-data`
```

File: Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        guard let data = Loader.contentsOf(fileNameWithDataEnvelope) else {
+            XCTFail(fileNameWithDataEnvelope + ".json not found")
+            return
+        }
+
+        // When
+        let statuses = try? mapper.map(response: data)
+
+        // Then
+        XCTAssertNotNil(mapper)
```
Comment (jaclync) [inline Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift]: any reasons asserting on the mapper? the mapper initialization doesn't seem nullable 🤔

File: Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    func test_order_statuses_is_mapped_from_response_with_data_envelope() {
+        // Given
+        let mapper = ReportOrderTotalsMapper(siteID: dummySiteID)
+        guard let data = Loader.contentsOf(fileNameWithDataEnvelope) else {
+            XCTFail(fileNameWithDataEnvelope + ".json not found")
+            return
+        }
+
+        // When
+        let statuses = try? mapper.map(response: data)
```
Comment (jaclync) [inline Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift]: nit: can probably make the result non-nil by using `try XCTUnwrap`

File: Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        // Then
+        XCTAssertNotNil(mapper)
+        XCTAssertEqual(statuses?.count, 8)
+    }
+}
+
+private extension ReportOrderTotalsMapperTests {
+    /// Returns the CustomerMapper output upon receiving `filename` (Data Encoded)
+    ///
+    func mapStatus(from filename: String) throws -> [OrderStatus] {
```
Comment (jaclync) [inline Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift]: nit: is this meant to be used?

**Review by jaclync — APPROVED** [review]
LGTM :shipit: there are some failures from related entities like refunds, products, and countries, I'm sure they will be handled separately.

File: Networking/NetworkingTests/Mapper/OrderListMapperTests.swift (hunk 1 above)
Comment (itsmeichigo) [reply Networking/NetworkingTests/Mapper/OrderListMapperTests.swift]: Updated in 7848519638b4e92116278ea02395cd4122d90974.

File: Networking/NetworkingTests/Mapper/OrderMapperTests.swift (hunk 2 above)
Comment (itsmeichigo) [reply Networking/NetworkingTests/Mapper/OrderMapperTests.swift]: Updated in 4d36b6c0b5bc68ed8e64894ee12394f7023b6d2b.

File: Networking/NetworkingTests/Mapper/OrderNotesMapperTests.swift (hunk 3 above)
Comment (itsmeichigo) [reply Networking/NetworkingTests/Mapper/OrderNotesMapperTests.swift]: Updated in f0f29442f88208e9a99e086b90d9f1ba8385cd6c.

File: Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift (hunk 4 above)
Comment (itsmeichigo) [reply Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift]: This is redundant indeed 😅 Removed in 6dad5975a61e0c23e254c79e1fc5d3129a99e851.

File: Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift (hunk 5 above)
Comment (itsmeichigo) [reply Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift]: Also updated in 6dad5975a61e0c23e254c79e1fc5d3129a99e851.

File: Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift (hunk 6 above)
Comment (itsmeichigo) [reply Networking/NetworkingTests/Mapper/ReportOrderTotalsMapperTests.swift]: This is also redundant - removed in 2f9d67d64ef6029f11b1ca73ec39673f3ce6ff27 👍

**Pull request merged**